src/Maestro/DependencyUpdater/DependencyUpdater.cs

```diff
@@ -203,6 +207,63 @@ select new
             }
         }
 
+        /// <summary>
+        ///     Update Longest Build Path table once a day for each channel at midnight
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        [CronSchedule("0 0 0 1/1 * ? *", TimeZones.PST)]
+        public async Task UpdateLongestBuildPathAsync(CancellationToken cancellationToken)
+        {
+            IQueryable<Channel> query = Context.Channels;
+
+            List<Channel> channels = query.AsEnumerable().Select(c => new Channel() { Id = c.Id, Name = c.Name }).ToList();
+
+            // Get the flow graph
+            var barOnlyRemote = await RemoteFactory.GetBarOnlyRemoteAsync(Logger);
+
+            List<Microsoft.DotNet.Maestro.Client.Models.DefaultChannel> defaultChannels = (await barOnlyRemote.GetDefaultChannelsAsync()).ToList();
+            List<Microsoft.DotNet.Maestro.Client.Models.Subscription> subscriptions = (await barOnlyRemote.GetSubscriptionsAsync()).ToList();
+
+            IEnumerable<string> frequencies = 
+                new string[] { "everyWeek", "twiceDaily", "everyDay", "everyBuild", "none", };
```

Comment:
>string [](start = 20, length = 6)

You can remove the "string" here and have it just be `new []{ "everyWeek", "twiceDaily", ... }`.

It's also a touch weird to store it as IEnumerable<string> instead of string[]

Reply:
The function it's passed to takes an IEnumerable

Reply:
You can pass the array to the function still.  It's not a big deal, was just surprising.  It's unusual to declare a local variable of a less specific type then it is (and usually there is a reason... some sort of weird cast or explicit interface method you are trying to get at).  No need to change it, mostly I was just commenting. :-)